data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/security/KubernetesCredentials.java

```diff
@@ -155,14 +155,15 @@ public String SASLMechanism() {
     if (SASLMechanism == null) {
       final var SASLMechanism = secret.getData().get(SASL_MECHANISM);
       if (SASLMechanism == null) {
-        return null;
+        this.SASLMechanism  = "PLAIN"; //default when not specified
```

Comment:
could we add some in `KubernetesCredentialsTest.java`?

Reply:
Yes going to add tests

Reply:
Added tests